### PR TITLE
[KYUUBI #4111] JDBC ExecuteStatement operation should contain operationLog

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

Close https://github.com/apache/kyuubi/issues/4111, JDBC ExecuteStatement operation should contain operationLog
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by Yikf** [comment]
cc @zhaomin1423

**Review by zhaomin1423 — APPROVED** [review]
LGTM

**Review by turboFei — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala [hunk 1]
```diff
@@ -142,8 +143,12 @@ abstract class OperationManager(name: String) extends AbstractService(name) {
       order: FetchOrientation,
       maxRows: Int): TRowSet = {
     val operationLog = getOperation(opHandle).getOperationLog
-    operationLog.map(_.read(maxRows)).getOrElse {
-      throw KyuubiSQLException(s"$opHandle failed to generate operation log")
+
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala:146]: nit: not needed line

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala [hunk 2]
```diff
@@ -142,8 +143,12 @@ abstract class OperationManager(name: String) extends AbstractService(name) {
...
+    operationLog match {
+      case Some(_) => operationLog.map(_.read(maxRows)).getOrElse {
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationManager.scala:148]:
```
case Some(log) => log.read(maxRows)
```

**Review by turboFei — APPROVED** [review]

**Comment by yaooqinn** [comment]
Hmm, isn't the error handled when an operation log is created?

**Comment by yaooqinn** [comment]
the original intentions are:
1. the operations that need operation logs shall implement getOperationLog
2. the client will only call fetching log for those operations which implement getOperationLog
3. the error raise only when the operation implements getOperationLog but fails to create the operation log file

**Comment by zhaomin1423** [comment]
> the original intentions are:
> 
> 1. the operations that need operation logs shall implement getOperationLog
> 2. the client will only call fetching log for those operations which implement getOperationLog
> 3. the error raise only when the operation implements getOperationLog but fails to create the operation log file

2 can't work well now.

**Comment by Yikf** [comment]
Yeah, 2 can not work well；

End-user can fetch logs through `OperationResource` as well, The client will only can fetch log for those operations which implement `getOperationLog` seems impossible to limit?

**Comment by yaooqinn** [comment]
After this change, we can not determine whether the log is failed or just not implemented

**Comment by yaooqinn** [comment]
> 2 can't work well now.

**This is by contract**, these OperationResouce's clients shall be able to benefit from the current implementation too.

**Comment by Yikf** [comment]
> After this change, we can not determine whether the log is failed or just not implemented

Good point, Have a question is which of the following designs is more reasonable:
1. Fetch empty log if the operation don't implements getOperationLog
2. Have the contract, client will only call fetching log for those operations which implement getOperationLog

If 1, we can handle whether the log is failed or just not implemented; 
if 2, we can handle this case according to this contract at client/jdbc engine side;

> shall be able to benefit from the current implementation too.

What are the benefits? (sorry for I didn't know the previous design ) : )

**Comment by yaooqinn** [comment]
> What are the benefits? (sorry for I didn't know the previous design ) : )

Well, the thrift and rest calls have no differences on both the log API and the backend operations. It's up to much higher level APIs(such as JDBC), or end-users, to do the error handling.

Currently, the meta operations are no operation logs and are executed synchronously. The rest call can also invoke these operations synchronously, which makes it has no chance to call log fetching

**Comment by Yikf** [comment]
I fix this issue at jdbc engine side, please take a look if you find a moment, thanks @yaooqinn @turboFei @zhaomin1423

**Review by yaooqinn — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

**Review by zhaomin1423 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**